apps/video/test/marionette/lib/video.js

```diff
@@ -147,6 +216,23 @@ Video.prototype = {
    return this.client.findElements(Video.Selector.thumbnail);
   },
 
+  get thumbnails() {
+    return this.client.findElements(Video.Selector.thumbnail);
+  },
+
+  waitForThumbnails: function(num) {
+    var thumbnails = this.thumbnails;
+
+    this.client.helper.waitFor(function() {
+      return thumbnails.length === num;
+    });
```

Comment:
You query the thumbnails once, get an array of them, and then wait for the length of that array to change.  As far as I can tell, thumbnails.length is never going to change. If it isn't === num before you call waitFor, I can't see how it is going to change while you wait. Maybe I don't understand marionette, but it seems to me that you need to be calling findElements() inside of the waitFor function.